src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1465,4 +1469,36 @@ public static MobEffect loadMobEffect(CompoundTag nbt, String key, @Nullable Mob
         }
     }
 
+    private static final ThreadLocal<RegistryResourceAccess> registryResouceAccess = new ThreadLocal<>();
+    private static final ThreadLocal<RegistryAccess> registryAccess = new ThreadLocal<>();
+
+    public static RegistryAccess.Writable createAndCacheWritable(ResourceManager manager) {
+        RegistryResourceAccess resourceAccess = RegistryResourceAccess.forResourceManager(manager);
+        RegistryAccess.Writable writable = RegistryAccess.builtinCopy();
+
+        //Register the default DimensionTypes
+        RegistryLoader loader = new RegistryLoader(resourceAccess);
+        WritableRegistry<DimensionType> registry = writable.ownedWritableRegistryOrThrow(Registry.DIMENSION_TYPE_REGISTRY);
+        DataResult<? extends Registry<DimensionType>> result = loader.overrideRegistryFromResources(registry, registry.key(), DimensionType.DIRECT_CODEC, RegistryOps.create(JsonOps.INSTANCE, writable));
+        result.error().ifPresent(err -> LOGGER.error("Unable to load Modded Dimension Types: {}", err.message()));
+
+        registryResouceAccess.set(resourceAccess);
+        registryAccess.set(writable);
+        return writable;
+    }
+
+    //Called from the tail of DimensionType#defaultDimensions
+    public static void registerDefaultDimensions(WritableRegistry<LevelStem> writableregistry) {
+        //This will always return unless #createAndCacheWritable has been called (we're on a dedicated server creating a new world)
+        if(registryAccess.get() == null || registryResouceAccess.get() == null) {
```

Comment:
Formatting: space before an `if`-expression's opening parenthesis, and braces on newlines.
```suggestion
        if (registryAccess.get() == null || registryResouceAccess.get() == null)
        {
```